File: LeetCode/00520. Detect Capital/00520.go
// https://leetcode.com/problems/detect-capital/submissions

func detectCapitalUse(word string) bool {
    if len(word) == 1 {
        return true;
    }

    var isUpper = func(c byte) bool { return 'A' <= c && c <= 'Z' }

    first, second := isUpper(word[0]), isUpper(word[1])

    if !first && second {
        return false
    }

    for i := 2; i < len(word); i++ {
        if isUpper(word[i]) != second {
            return false
        }
    }

    return true
}

File: LeetCode/00520. Detect Capital/00520.c
// https://leetcode.com/problems/detect-capital

bool isUpper(int c) {
    return 'A' <= c && c <= 'Z';
}

bool detectCapitalUse(char *word) {
    int len = strlen(word);
    
    if (len == 1) {
        return true;
    }

    bool first = isUpper(word[0]);
    bool second = isUpper(word[1]);

    if (!first && second) {
        return false;
    }

    for (int i = 2; i < len; ++i) {
        if (isUpper(word[i]) != second) {
            return false;
        }
    }

    return true;
}
